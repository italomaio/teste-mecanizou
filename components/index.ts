export { default as Card } from "./Card/Card";
export { default as RootProviders } from "./RootProviders/RootProviders";
export { default as Logo, type LogoProps } from "./Logo/Logo";
export { default as Input } from "./Input/Input";
export { default as Label, type LabelProps } from "./Label/Label";
export {
  default as Button,
  type ButtonVariant,
  type ButtonSize,
  type ButtonProps,
} from "./Button/Button";
export { default as ThemeToggle } from "./ThemeToggle/ThemeToggle";
export { default as ProductLink } from "./ProductLink/ProductLink";
export { default as Navbar } from "./Navbar/Navbar";
export { default as Skeleton } from "./Skeleton/Skeleton";
export { default as Pagination } from "./Pagination/Pagination";
export { default as ProductRating } from "./ProductRating/ProductRating";
export { default as ProductCard } from "./ProductCard/ProductCard";
export { default as ProductQuantity } from "./ProductQuantity/ProductQuantity";
export { default as ProductImages } from "./ProductImages/ProductImages";
export { default as ProductReviews } from "./ProductReviews/ProductReviews";
export { default as Stars } from "./ProductRating/Stars";
export { default as Loader } from "./Loader/Loader";
export { default as ProductPrices } from "./ProductPrices/ProductPrices";
export {
  default as UserDropdown,
  type UserDropdownProps,
} from "./UserDropdown/UserDropdown";
