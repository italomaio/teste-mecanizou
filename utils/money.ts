export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
