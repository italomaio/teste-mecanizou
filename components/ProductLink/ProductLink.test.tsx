import { render, screen } from "@/tests/utils";
import { ProductLink } from "@/components";

describe("ProductLink", () => {
  it("Should render correctly", () => {
    const { container } = render(<ProductLink>Test</ProductLink>);
    expect(container).toBeInTheDocument();
  });

  it("Should render with correct href when productId is provided", () => {
    render(<ProductLink productId="1">Test</ProductLink>);
    const link = screen.getByRole("link", { name: /View product 1/i });
    expect(link).toHaveAttribute("href", "/products/1");
  });

  it("Should render with correct href when productId is not provided", () => {
    render(<ProductLink>Test</ProductLink>);
    const link = screen.getByRole("link", { name: /Go to reviews/i });
    expect(link).toHaveAttribute("href", "#reviews");
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(
      <ProductLink productId="1">Test</ProductLink>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
