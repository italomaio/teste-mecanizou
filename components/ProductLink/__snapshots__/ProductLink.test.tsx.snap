// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductLink Should match snapshot 1`] = `
<DocumentFragment>
  <a
    aria-label="View product 1"
    class="focus-visible-ring flex flex-col"
    href="/products/1"
  >
    Test
  </a>
</DocumentFragment>
`;
